content/chapters/compute/lecture/slides.mdpp

```diff
@@ -811,9 +796,16 @@ var = 2000000; time = 1268 ms
 
 ----
 
+## Locks vs Notifications
+
+![Lock vs Notification](./media/lock-vs-notify.svg)
+
+----
+
 ### Semaphore
 
 * Generalized mutex (mutex = binary semaphore)
+* Multiple threads are in the critical section at the same time
```

Comment:
No. A critical section is a piece of code that threads should only access concurrently (one at a time). This is achieved via mutexes, not semaphores.

Reply:
On the next slide, when comparing spinlocks, mutexes, and semaphores based on the number of threads that can enter the critical section we say `Multiple threads can enter critical section` for semaphores.
Should this not be changed as well?

Reply:
Yeah, it should.

Reply:
I removed my addition for now.